routes/products.js

```diff
@@ -0,0 +1,133 @@
+const express = require("express");
+const db = require("../database");
+
+const router = express.Router();
+
+router.get("/:product_id", (req, res) => {
+  db("products")
+    .where({ id: req.params.product_id })
+    .then(results => {
+      if (results.length === 0) {
+        res.status(500).json({ message: "Product not found" });
+      } else {
+        res.json(results[0]);
+      }
+    })
+    .catch(err => {
+      res.status(500).json({ message: err.message });
+    });
+});
+
+router.get("/", (req, res) => {
+  db("products")
+    .then(results => {
+      res.json(results);
+    })
+    .catch(err => {
+      res.status(500).json({ message: err.message });
+    });
+});
+
+router.post("/new", (req, res) => {
+  db("products")
+    .where({ title: req.body.title })
+    .then(results => {
+      if (results.length > 0) {
+        res.status(500).json({ message: "Product already exists" });
+      } else if (
+        !req.body.title ||
+        !req.body.description ||
+        !req.body.inventory ||
+        !req.body.price
+      ) {
+        res.status(500).json({ message: "Must POST all product fields" });
+      } else {
+        db("products")
```

Comment:
Do not nest promises, return them instead so that they are part of the parent promise.